chapter0-What-is-Java/terminology.md

```diff
@@ -1,19 +1,21 @@
 
-**Linker:** A collection of object files are linked together to create an executable file (i.e., the operating system can load the executable into RAM to run the program). Another term for an executable is "(relocatable) machine code".
+**Platform** A specific target machine that code can be executed on. For example, if you compile a C++ program on a Windows machine, the executable file can be copied to any other machine but it will only run on other Windows machines but never another machine (e.g., a Mac or a Linux machine). A platform is determined by the target machine's hardware and operating system.  Different processor types, for example, will utilise different codes for the same operations.
 
-**Object File:** The result from compiling source code. An object file isn't easily read by a human but it may not be runnable on a computer. For example, if your program takes a square root of a number, your program will rely on the mathematical program (provided by the math library of the language) that actually determines how to compute a square root. The object file for the program will refer to the square root but will not have the code explaining how the square root computation works.
+**Object File:** The result from compiling source code. Object files can have different formats and may also depend on other object files or programs.  Hence, some will not be directly executable without further compilation and / or in isolation.  Several object files, which depend on each other to create a fully-functioning program, can be linked together to create an executable file.
 
-**Interpreter** An interpreted language takes each high-level statement, determines its low-level version and executes (while linking if need be) the result. This is done for each statement in succession (before the next high-level statement is even looked at). This  conversion is "on-the-fly", also called "as needed".
+**Linker:** Links a collection of object files to form an executable file.
+
+**Executable file:** A file that the operating system can load into memory and run directly without further compilation.  Executable files are sometimes known as 'relocatable' since they can be moved to and run on any instance of the intended platform (see 'platform', below).
 
-**Bytecode** The result from the Java compiler (often called the JVM compiler). Essentially it is an object file for a virtual machine, which allows it to be used on any machine that has Java. (However, bytecode is not an executable file.  To execute a bytecode file, you need to invoke a Java interpreter).
+**Interpreter** An interpreted language takes each high-level statement, determines its low-level version and executes (while linking if need be) the result. This is done for each statement in succession (before the next high-level statement is even looked at). This  conversion is "on-the-fly", also called "as needed".
```

Comment:
Similarly to a compiler, an interpreter turns high-level code into instruction the computer can execute. 
However, that translation is happening for each statement in succession (before the next high-level statement is even looked at).
This conversion is called "on-the-fly" or "as needed".

Reply:
👍